query/define.rs

```diff
@@ -6,107 +6,606 @@
 
 use std::{error::Error, fmt};
 
-use concept::{error::ConceptWriteError, type_::type_manager::TypeManager};
-use encoding::value::label::Label;
-use storage::snapshot::WritableSnapshot;
+use answer::Type as TypeEnum;
+use concept::{
+    error::{ConceptReadError, ConceptWriteError},
+    type_::{
+        annotation::{Annotation, AnnotationError},
+        attribute_type::AttributeTypeAnnotation,
+        entity_type::EntityTypeAnnotation,
+        object_type::ObjectType,
+        owns::OwnsAnnotation,
+        plays::PlaysAnnotation,
+        relates::RelatesAnnotation,
+        relation_type::RelationTypeAnnotation,
+        type_manager::TypeManager,
+        KindAPI, Ordering, OwnerAPI, PlayerAPI,
+    },
+};
+use encoding::{
+    graph::type_::Kind,
+    value::{label::Label, value_type::ValueType},
+};
+use storage::snapshot::{ReadableSnapshot, WritableSnapshot};
 use typeql::{
     common::token,
     query::schema::Define,
-    schema::definable::{Struct, Type},
-    Definable,
+    schema::definable::{
+        struct_::Field,
+        type_::{Capability, CapabilityBase},
+        Struct, Type,
+    },
+    type_::{NamedType, Optional},
+    Definable, ScopedLabel, TypeRef, TypeRefAny,
+};
+
+use crate::{
+    util::{resolve_type, resolve_value_type, translate_annotation},
+    SymbolResolutionError,
 };
 
+macro_rules! unwrap_or_else {
+    ($variant:path = $to_unwrap:expr ; $else_:block) => {
+        if let $variant(inner) = $to_unwrap {
+            inner
+        } else {
+            $else_
+        };
+    };
+}
```

Comment:
This should be deleted and the usages replaced with `let-else`:
```rs
let $variant(inner) = $to_unwrap else { return $else_; };
```
or
```rs
let $variant(inner) = $to_unwrap else { continue; };
```
(the inside of the `else {}` must diverge, i.e. `return`, `panic!`, `break`, or `continue`)